Finalise all low bit rate Capture App with x64

Brumi-2021 opened the pull request:
Hello , in this PR , we activated the x64  and /64 , and we could finalize the pending low bit rate REC in Capture App,below 75khz   (cont. of yesterday PR , #1380. 

So now , thanks to the dynamic oversampling / decimation , we managed to have a correct REC functionality 
for this range  12k5 .... 1Mhz in C16 / C8 format. 

Initially I thought that we will need to go to x128 ,  but finally it is not need it . 

I evaluated now all range , and made fine tuning about the freq, and the needed sample rate...
As you could imagine it is not simple because we should select the best trade off , of many issues , in terms of achieve
* reasonable "moirée effect" or vertical stripes  in the screen - but easy to solve just make fine tuning +-10Hz center freq. , 
* good enough anti-aliasing filters (no background noise),
* good RF image reject   (or at least acceptable ) 
* good M4 usage % balance, in the 1st DFU DEBUG overlay table ,
* good enough ADC sample rate,

It is functional and I tested Capturing and Replay all low bit rates from 250k , 150k, 100k, 75k, 50k, 32k, 16k,12k5 
and it works well , (same time duration as recorded , and good quality reproduction in a receiver, good coherent recorded file size ,  and good display with inspectrum . 

Just  screenshots of the tested AM  , MF  upconverted captured signals  (sorry I missed the 100k picture but also tested) 
![image](https://github.com/eried/portapack-mayhem/assets/86470699/34b10075-e070-4154-8296-88c4bef5e85a)

 Example  recorded BW 16K ,  with 10 sec, and inspected by inspectrum 
![image](https://github.com/eried/portapack-mayhem/assets/86470699/b4b74d4d-4d48-440c-b3a5-369ac53816db)


It is pending to apply what  @NotherNgineer and @kallanreed  suggested yesterday in my previous PR , but we agreed to finalize before  the hole series of oversampling / decimators and its fine tuning ,  before ,( x8, x16, x32, x64.) (done in this PR)


Notes : 
1-In the Replay List app,   if combining many play files , in the low bit rate, sometimes there is a bug in the screen , showing veritcal stripes , but then replaying the same file isolatedly it works well ,
2-) In the Replay sometimes Hackrf is radiating a captured DC spike in 0Hz,  so when re-radiating it , take into account that it could produce a strange tone in the receiver .  (that can be easily avoid it , just recording with higher BW and apply some offset.

kallanreed commented on firmware/common/oversample.hpp:
```diff
@@ -57,10 +57,9 @@
-    if (sample_rate < 100'000) return OversampleRate::x64;  // 50k, prepared for future, OVS ok, but decim. x64 still not implemented.
-    if (sample_rate < 150'000) return OversampleRate::x32;  // 100k needs x32
-    if (sample_rate < 250'000) return OversampleRate::x16;  // 150k needs x16
+    if (sample_rate < 25'000) return OversampleRate::x64;   // 16k , 12k5.
```
Looks like you changed 25K to be 32K in the SPEC list. Should you change the threshold on line 60?

kallanreed replied:
I'd like it if these values were kept parallel with the SPEC bandwidths. There are so many magic numbers already, it would help make it easier to see where these values are coming from.

NotherNgineer reviewed (approved):
LGTM

Brumi-2021 commented:
As I do not see my just added comments  , let me add it  here again.

(Regarding deleted BW 25K  in my initial PR submission)
In previous PR , I removed intentionally the BW option 25k , because we have to select among discrete OVS values , 
and with x32 OVS it was too poor performances , and with x64 good performances but , it was too easy to get too much "vertical stripes effect" in the initial tunning . (that can be easily addressed by re-adjusting the tunning by stepss of +-10Hz).

Anyway , based on your comment ,  as I think it is better to offer to the user more BW Options ,  I just added back again the BW=25K option , then user will be able to select the most suitable option to his needs,   (if later on we consider that we have too much BW options , we can easily remove the 600k or 650k or both, because I just added 1 year ago , as a debugging purposes).

![image](https://github.com/eried/portapack-mayhem/assets/86470699/93f52426-5847-4bdc-b0a1-3dd6c64852b2)

gullradriel reviewed (approved):
LGTM too
